src/rt/config.d

```diff
@@ -58,7 +58,7 @@ import core.stdc.string : strlen;
 
 extern extern(C) string[] rt_args() @nogc nothrow;
 
-alias rt_configCallBack = string delegate(string) @nogc nothrow;
+alias rt_configCallBack = string delegate(return scope string) @nogc nothrow;
 
```

Comment:
You probably also need to update https://github.com/dlang/druntime/blob/v2.073.0-b1/src/gc/config.d#L43